sdk/storage/Azure.Storage.Blobs/src/Models/BlobInfo.cs

```diff
@@ -237,6 +237,16 @@ public partial class BlobDownloadDetails
         /// If this blob is sealed.
         /// </summary>
         public bool IsSealed => _flattened.IsSealed;
+
+        /// <summary>
+        /// x-ms-or
+        /// </summary>
+        public IDictionary<string, IDictionary<string, string>> ObjectReplicationSourceProperties => BlobExtensions.ParseObjectReplicationIds(_flattened.ObjectReplicationRules);
```

Comment:
Should we do the parsing at construction of this object? current way would trigger parsing everytime someone is accessing property.

Reply:
See lines 83 and 247 of BlobInfo.cs for changes